src/utils/IstioConfigUtils.ts

```diff
@@ -64,3 +64,39 @@ export const getIstioObject = (istioObjectDetails?: IstioConfigDetails) => {
   }
   return istioObject;
 };
+
+const nsRegexp = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[-a-z0-9]([-a-z0-9]*[a-z0-9])?)*$/;
+const hostRegexp = /(?=^.{4,253}$)(^((?!-)(([a-zA-Z0-9-]{0,62}[a-zA-Z0-9])|\*)\.)+[a-zA-Z]{2,63}$)/;
+
+// Used to check if Sidecar and Gateway host expressions are valid
+export const isServerHostValid = (serverHost: string): boolean => {
```

Comment:
I wonder if we should not relax a little bit the hosts validation. If I understand correctly, when hosts are detected as invalid we will disable the "create" button, right? That makes us very dependant on these regexes and on any special handling done by Istio or not (e.g. wildcards). What if we display the warning message but without blocking creation?

Reply:
Main problem is that Istio is quite strict on creation here, so my first version addressed the relaxed version, but it catches back the Istio error.
So, now the logic is mapped with the Istio requeriment for this hosts, where also the wildcards supported by Istio are included in the validation logic.
One obvious case is that, for example, Istio doesn't permit for these objects to create a "short" name, so I think is better that Kiali should allow user to add the correct pattern.

Give it a try, perhaps you see something I missed, but I think it's a 1:1 in the expected format.
Note that this helper method doesn't cover all validation, some of full validation are in the Gateway and Sidecar specific forms.